WhitespaceAfter should support LITERAL_YIELD token

https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac WithYield.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_YIELD"/>
        </module>
    </module>
</module>

/var/tmp $ cat WithYield.java
public class WithYield {
    public static void main(String[] args) {
        int a = switch (args[0]) {
            case "got":
                yield 1;
            case "my":
                yield(3);
            default:
                yield 2;
        };
        System.out.println(a);
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml WithYield.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_YIELD" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_YIELD" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `yield` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_YIELD`

#11489, #11493 and #11494 are similar requests for `try`/`finally`/`synchronized` keywords support

```
public class InputWhitespaceAfterLiteralYield {
   public static void main(String[] args) {
     int a = switch (args[0]) {
       case "got":
         yield (1); // OK
       case "my":
         yield(3); // violation ''yield' is not followed by whitespace'
     };
   }
}
```
I am using above code in input file but I am getting this error- 
[ERROR]/C:/Users/Dell/OneDrive/Desktop/G_Checkstyle/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java:[12,14] switch expressions are not supported in -source 11
[ERROR]   (use -source 14 or higher to enable switch expressions)
I changed settings also but it didn't work
![image](https://user-images.githubusercontent.com/78761614/166798126-44f7c632-a47a-48b5-9874-11c8c52a8378.png)
Please guide.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -58,6 +58,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * LITERAL_FINALLY</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_RETURN">
  * LITERAL_RETURN</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_YIELD">
+ * LITERAL_YIELD</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_CATCH">
  * LITERAL_CATCH</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
@@ -118,6 +120,16 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *  }
  *  public String testTwo() {
  *      return("a" + "b"); // violation 'return' is not followed by whitespace'
+ *  }
+ *  public static void main(String[] args) {
+ *    int a = switch (args[0]) {
+ *      case "got":
+ *        yield (1); // OK
+ *      case "my":
+ *        yield(3); // violation ''yield' is not followed by whitespace'
+ *      default:
+ *        yield 2;
+ *    };
  *  }
  * </pre>
  * <p>
@@ -191,6 +203,7 @@ public class WhitespaceAfterCheck
             TokenTypes.LITERAL_FOR,
             TokenTypes.LITERAL_FINALLY,
             TokenTypes.LITERAL_RETURN,
+            TokenTypes.LITERAL_YIELD,
             TokenTypes.LITERAL_CATCH,
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -112,8 +112,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *      try {} catch (Exception e){} // OK
  *      try {} catch(Exception e){} // violation ''catch' is not followed by whitespace'
  *
+ *      synchronized (this) { } // OK
  *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
- *      synchronized (this) { } // ok
  *  }
  *  public String testOne() {
  *      return ("a" + "b"); // OK

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,LITERAL_FINALLY,LITERAL_RETURN,LITERAL_CATCH,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_SYNCHRONIZED,LITERAL_TRY,LAMBDA"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,LITERAL_FINALLY,LITERAL_RETURN,LITERAL_YIELD,LITERAL_CATCH,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_SYNCHRONIZED,LITERAL_TRY,LAMBDA"
                        name="tokens"
                        type="java.lang.String[]"
                        validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -125,7 +125,8 @@
                     NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION, RCURLY, SL, SLIST, SL_ASSIGN, SR,
                     SR_ASSIGN, STAR, STAR_ASSIGN, LITERAL_ASSERT, TYPE_EXTENSION_AND"/>
       <message key="ws.notFollowed"
-              value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
+              value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks
+               may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
       <message key="ws.notPreceded"
               value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
     </module>

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -106,7 +106,8 @@
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE, LITERAL_RETURN,
                     LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
-                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA"/>
+                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA,
+                    LITERAL_YIELD"/>
     </module>
     <module name="WhitespaceAround">
       <property name="allowEmptyConstructors" value="true"/>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -163,6 +163,16 @@ public class WhitespaceAfterCheckTest
     }
 
     @Test
+    public void testLiteralYield() throws Exception {
+        final String[] expected = {
+            "17:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "yield"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputWhitespaceAfterLiteralYield.java"),
+                expected);
+    }
+
+    @Test
     public void testLiteralSynchronized() throws Exception {
         final String[] expected = {
             "13:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "synchronized"),

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java
@@ -1,0 +1,22 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_YIELD
+
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterLiteralYield {
+  public static void main(String[] args) {
+    int a = switch (args[0]) {
+      case "got":
+        yield (1); // OK
+      case "my":
+        yield(3); // violation ''yield' is not followed by whitespace'
+      default:
+        yield 2;
+    };
+  }
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2904,6 +2904,8 @@ class Bar {
                   LITERAL_FINALLY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_RETURN">
                   LITERAL_RETURN</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_YIELD">
+                  LITERAL_YIELD</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_CATCH">
                   LITERAL_CATCH</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
@@ -2942,6 +2944,8 @@ class Bar {
                   LITERAL_FINALLY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_RETURN">
                   LITERAL_RETURN</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_YIELD">
+                  LITERAL_YIELD</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_CATCH">
                   LITERAL_CATCH</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
@@ -3010,6 +3014,16 @@ public String testOne() {
 }
 public String testTwo() {
     return("a" + "b"); // violation 'return' is not followed by whitespace'
+}
+public static void main(String[] args) {
+  int a = switch (args[0]) {
+    case "got":
+      yield (1); // OK
+    case "my":
+      yield(3); // violation ''yield' is not followed by whitespace'
+    default:
+      yield 2;
+  };
 }
           </pre>
         </div>

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -3006,8 +3006,8 @@ public void myTest() {
     try {} catch (Exception e){} // OK
     try {} catch(Exception e){} // violation ''catch' is not followed by whitespace'
 
+    synchronized (this) { } // OK
     synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
-    synchronized (this) { } // ok
 }
 public String testOne() {
     return ("a" + "b"); // OK

